### ISSUE TITLE
Guitarix UI cannot be opened in Carla

### ISSUE THREAD

**Issue by paoloperrone** [falkTX/Carla#1445]
Carla-git version 2.3.0
Guitarix 0.42.1 (installed from source)

Both applications work well separately. If I load Guitarix in Carla as a Jack application, the sound work, but if I click on the gear icon to load the UI, Guitarix gets deactivated by Carla (it seems to "crash" internally to Carla, in some sense). 

If I check the log, during runtime of Guitarixm the following line gets repeated continuously in the Carla log, flooding it: 
```
Carla assertion failure: "jbasebuf->isValid" in file libjack_midi.cpp, line 30
```

But mostly, as soon as try to load the Guitarix UI the log says:
```
(guitarix:5782): Gdk-ERROR **: 16:42:47.690: The program 'guitarix' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadWindow (invalid Window parameter)'.
  (Details: serial 510 error_code 3 request_code 8 (core protocol) minor_code 0)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the GDK_SYNCHRONIZE environment
   variable to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
waitForClient(process) timed out
```

**Comment by falkTX** [comment]
There are some flags you need to set for guitarix for that to work well.
I think -E to use a single client, not sure the rest.

**Comment by paoloperrone** [comment]
Yep, I use the flags -D and -E. 
(The problem seems to persist, no matter the flags, except of course for the flag -N)

**Comment by SanderVocke** [comment]
I am having the same issue with guitarix -D -E.

**Comment by falkTX** [comment]
Fixed in 4fac74406762a933b6b0a87297995a40c6e693df
